### PR TITLE
0x5a GAS

### PR DESCRIPTION
**Pull request by danilowhk**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-0x5a GAS opcode
-
-

### CODE REVIEW

File: src/kakarot/execution_context.cairo [hunk 1]
```diff
@@ -38,7 +38,7 @@ namespace ExecutionContext {
         let initial_pc = 0;
         let gas_used = 0;
         // TODO: Add support for gas limit
-        let gas_limit = 0;
+        let gas_limit = 21002;
```
Comment (abdelhamidbakhta) [inline src/kakarot/execution_context.cairo:41]: Why do you hardcode the gas limit ? It should come from the user. We need to redefine a bit the execution context to add some data from the transaction. Please put on hold this PR for the moment

**Review by abdelhamidbakhta — CHANGES_REQUESTED** [review]
Put on hold because we need to think about how we design gas limit and transaction parameters from user

File: src/kakarot/execution_context.cairo (hunk 1 above)
Comment (danilowhk) [reply src/kakarot/execution_context.cairo:41]: Ok! I added the gas limit just to test the (Gas_limit - gas_used -2) , when it was gas_limit = 0 it was giving and overflow error

Comment (danilowhk) [reply src/kakarot/execution_context.cairo:41]: Should i close it for now then?

Comment (abdelhamidbakhta) [reply src/kakarot/execution_context.cairo:41]: We can let it open. I will put a special status with a label

Comment (abdelhamidbakhta) [reply src/kakarot/execution_context.cairo:41]: So that you can resume the work once we design a solution. Thanks for your help

Comment (danilowhk) [reply src/kakarot/execution_context.cairo:41]: Ok! Will make the changes later then :)

**Pull request closed without merging**